.github/workflows/roll-pinned-toolchain-versions.yml

```diff
@@ -0,0 +1,93 @@
+# still fail CI. In particular, some nightly releases do not support all of the
+# target architectures that we use in CI; attempting to roll to any such release
+# will fail.
+
+name: Roll pinned toolchain versions
+on:
+  schedule:
+    - cron: '29 12 * * *'
+
+permissions: read-all
+
+jobs:
+  roll:
+    name: Roll pinned toolchain versions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+        with:
+          ref: main
+          persist-credentials: false
+      - name: Calculate target nightly version
+        # Use yesterday's date (`-d '-1 day'`) so we're sure the nightly for that
+        # date has actually been published yet. This allows us to not worry
+        # about what time of day this job runs.
+        run: echo "ZC_TARGET_NIGHTLY=nightly-$(date -d '-1 day' +%Y-%m-%d)" >> $GITHUB_ENV
+      - name: Install Rust with ${{ env.ZC_TARGET_NIGHTLY }} toolchain
+        uses: dtolnay/rust-toolchain@00b49be78f40fba4e87296b2ead62868750bdd83 # stable
+        with:
+            toolchain: ${{ env.ZC_TARGET_NIGHTLY }}
+        # Install whatever the latest stable release is. This has the side
+        # effect of determining the latest stable release so that we can update
+        # `Cargo.toml`.
+      - name: Install Rust with stable toolchain
+        uses: dtolnay/rust-toolchain@00b49be78f40fba4e87296b2ead62868750bdd83 # stable
+        with:
+            toolchain: stable
+      - name: Update files
+        run: |
+          set -eo pipefail
+
+          function validate-file {
+            REGEX="$1"
+            FILE="$2"
+            grep "$REGEX" "$FILE" >/dev/null || { echo "Failed to find line matching regex '$REGEX' in $FILE" >&2; exit 1; }
+          }
+
+          function update-pinned-version {
+            VERSION_NAME="$1"
+            VERSION="$2"
+            # For nightly, this is the same as `$VERSION`. For stable, it's
+            # `stable` because `rustup` doesn't recognize that `x.y.z` refers to
+            # the same thing as `stable` even if they're the same toolchain.
+            VERSION_FOR_CARGO="$3"
+            ZEROCOPY_FEATURES="$4"
+
+            # Confirm that `Cargo.toml` lists the pinned version in the expected
+            # format. This is a prerequisite for the subsequent `sed` command.
+            REGEX="^pinned-$VERSION_NAME = \"[a-z0-9\.-]*\"$"
+            validate-file "$REGEX" Cargo.toml
+            sed -i -e "s/$REGEX/pinned-$VERSION_NAME = \"$VERSION\"/" Cargo.toml
+
+            # Confirm that the update didn't bork `Cargo.toml`.
+            validate-file "$REGEX" Cargo.toml
+
+            # Update `.stderr` files as needed for the new version.
+            TRYBUILD=overwrite cargo "+$VERSION_FOR_CARGO" test --package zerocopy $ZEROCOPY_FEATURES
+            TRYBUILD=overwrite cargo "+$VERSION_FOR_CARGO" test --package zerocopy-derive
+          }
```

Comment:
Once the [cargo script](https://github.com/rust-lang/rfcs/pull/3502) work is stabilized, it might be a tad more robust to use `toml_edit` for this.

Reply:
Ah good idea!